src/core/thread/thread_netif.cpp

```diff
@@ -180,22 +179,24 @@ otError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Address
     return error;
 }
 
-otError ThreadNetif::TmfFilter(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo, void *aContext)
+otError ThreadNetif::ApplyTmfFilter(const Ip6::MessageInfo &aMessageInfo, Mle::MleRouter &aMle)
```

Comment:
Is it possible to involve a public non-static method like 
```cpp
bool IsTmfMessage(const Ip6::MessageInfo &aMessageInfo) const
```
and calling it both in `TmfFilter()` and IP layer's filtering function?
It looks strange passing `Mle` as parameter into a function of `ThreadNetif` which is actually owner of the `Mle`.